nautobot/core/views/mixins.py

```diff
@@ -903,3 +911,72 @@ def perform_bulk_update(self, request, **kwargs):  # pylint: disable=arguments-d
             return redirect(self.get_return_url(request))
         data.update({"table": table})
         return Response(data)
+
+
+class ObjectChangeLogViewMixin(NautobotViewSetMixin):
+    """
+    UI mixin to list a model's changelog queryset
+    """
+
+    base_template = None
+
+    def changelog(self, request, *args, **kwargs):
+        obj = self.get_object()
+        content_type = ContentType.objects.get_for_model(obj)
+        # v2 TODO(jathan): Replace prefetch_related with select_related
+        objectchanges = (
+            ObjectChange.objects.restrict(request.user, "view")
+            .prefetch_related("user", "changed_object_type")
+            .filter(
+                Q(changed_object_type=content_type, changed_object_id=obj.pk)
+                | Q(related_object_type=content_type, related_object_id=obj.pk)
+            )
+        )
+        objectchanges_table = ObjectChangeTable(data=objectchanges, orderable=False)
```

Comment:
Part of me wants to move this into the `renders.NautobotHTMLRenderer.construct_table` function, but this table deviates a little from that. What do y'all think? Would it be worth the effort?

Reply:
IMHO, I think we should try to move this into the `construct_table` function.
Just stealing from @jathanism: renderer should do all the heavy lifting and keep the mixins as lightweight as possible.

Reply:
Updated.